File: scripts/classic_aa_test.m
function matLatSam = classic_aa_test(matFeatSam)
% This function calls an R script to estimate H using existing archetypes
%
% Supporting function: classic_aa_test.R
% Intermediate files: classic_aa_test_Input.mat classic_aa_test_Output.mat (deleted after run by default)
% Archetypes are assumed to exist in bestModel.Rdat
%
% Note: change path to Rscript accordingly
%
% Copyright Sohan Seth sohan.seth@hiit.fi

save classic_aa_test_Input.mat matFeatSam
% In triton
% [status, msg] = system('/m/fs/software/R/ubuntu/R-2.15.1/bin/Rscript classic_aa.R');
% [status, msg] = system('Rscript classic_aa.R');
% if status ~= 0
%     disp(msg);
% end
!/opt/local/bin/Rscript classic_aa_test.R
load classic_aa_test_Output.mat matLatSam
!rm classic_aa_test_Input.mat classic_aa_test_Output.mat

File: scripts/classic_aa_test.R
#!usr/bin/env Rscript
# Supporting function for classic_aa_test.m
#
# Copyright Sohan Seth and Manuel Eugster

set.seed(1)

library("archetypes")
library(R.matlab)
load("bestModel.RData")

predict.archetypes <- 
function (object, newdata, ...)
{
   stopifnot(object$family$which == "original")
   scale <- object$scaling
   x <- t(newdata)
   x <- x - scale$mean
   x <- x/scale$sd
   x <- object$family$dummyfn(x, ...)
   zs <- t(parameters(object))
   zs <- zs - scale$mean
   zs <- zs/scale$sd
   zs <- rbind(zs, 200)
   alphas <- matrix(NA, ncol = ncol(x), nrow = ncol(coef(object)))
   alphas <- object$family$alphasfn(alphas, zs, x)
   t(alphas)
}

inp <- readMat('classic_aa_test_Input.mat')

op <- predict.archetypes(a, t(inp$matFeatSam))
writeMat('classic_aa_test_Output.mat', matLatSam = t(op))